rust/src/main.rs

```diff
@@ -156,13 +187,26 @@ fn main() -> anyhow::Result<()> {
             continue;
         }
 
-        let mut context = TestContext::new(&config.settings);
-        //TODO: AssertUnwindSafe should be used with caution
-        let mut ctx_wrapper = AssertUnwindSafe(&mut context);
+        let result = catch_unwind(move || {
+            match test_case.fun {
+                TestFn::NonSerialized(fun) => {
+                    let mut context = TestContext::new(&config.settings.naptime);
```

Comment:
Here and below you partially reverted PR #65, going back to initializing TestContext with a naptime, not a SettingsConfig.  Was that intentional or a merge error?

Reply:
Definitely an error, sorry...